bottlerocket-settings-models/settings-models/src/kubernetes/mod.rs

```diff
@@ -96,4 +97,36 @@ pub struct KubernetesSettingsV1 {
     hostname_override: ValidLinuxHostname,
     // Generated in `k8s-1.25+` variants only
     seccomp_default: bool,
+    #[cfg(feature = "nvidia-device-plugin")]
+    device_plugin: K8sDevicePluginSettingsV1,
+}
+
+// Define the DevicePlugin struct
+#[model(impl_default = true)]
+pub struct K8sDevicePluginSettingsV1 {
+    nvidia: NvidiaDevicePluginSettingsV1,
+}
+
+/// NvidiaRuntimeSettingsV1 contains the settings for a container runtime settings for Nvidia gpu.
+#[model(impl_default = true)]
+pub struct NvidiaDevicePluginSettingsV1 {
```

Comment:
This need not be versioned.

```suggestion
pub struct NvidiaDevicePluginSettings {
```

Reply:
renamed.